realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.transformer
+import com.android.SdkConstants
+import com.android.build.api.transform.*
+import com.google.common.collect.ImmutableSet
+import com.google.common.collect.Sets
+import groovy.io.FileType
+import io.realm.annotations.Ignore
+import javassist.ClassPool
+import javassist.LoaderClassPath
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.lang.reflect.Modifier
+import java.util.jar.JarFile
+
+import static com.android.build.api.transform.QualifiedContent.*
+/**
+ * This class implements the Transform API provided by the Android Gradle plugin
+ */
+class RealmTransformer extends Transform {
+
+    private Logger logger = LoggerFactory.getLogger('realm-logger')
+
+    @Override
+    String getName() {
+        return "RealmTransformer"
+    }
+
+    @Override
+    Set<ContentType> getInputTypes() {
+        return ImmutableSet.<ContentType> of(DefaultContentType.CLASSES)
+    }
+
+    @Override
+    Set<Scope> getScopes() {
+        return Sets.immutableEnumSet(Scope.PROJECT)
+    }
+
+    @Override
+    Set<Scope> getReferencedScopes() {
+        return Sets.immutableEnumSet(Scope.EXTERNAL_LIBRARIES, Scope.TESTED_CODE)
+    }
+
+    @Override
+    boolean isIncremental() {
+        return false
+    }
+
+    @Override
+    void transform(Context context, Collection<TransformInput> inputs, Collection<TransformInput> referencedInputs,
+                   TransformOutputProvider outputProvider, boolean isIncremental)
+            throws IOException, TransformException, InterruptedException {
+
+        def tic = System.currentTimeMillis()
+
+        // Find all the class names
+        def classNames = getClassNames(inputs)
+
+        // Create and populate the Javassist class pool
+        ClassPool classPool = createClassPool(inputs, referencedInputs)
+
+        logger.info "ClassPool contains Realm classes: ${classPool.getOrNull('io.realm.RealmList') != null}"
+
+        // Find the proxy classes
+        def proxyClasses = classNames
+                .findAll { it.endsWith('RealmProxy') }
```

Comment:
(Edge case) this is might bring some class ending with the suffix but not part of the model, can we use also 
```java
RealmObject.class.isAssignableFrom(clazz)
```